test/extensions/formatter/req_without_query/req_without_query_test.cc

```diff
@@ -0,0 +1,104 @@
+#include "envoy/config/core/v3/substitution_format_string.pb.validate.h"
+
+#include "common/formatter/substitution_format_string.h"
+
+#include "test/mocks/server/factory_context.h"
+#include "test/mocks/stream_info/mocks.h"
+#include "test/test_common/utility.h"
+
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace Formatter {
+
+class ReqWithoutQueryTest : public ::testing::Test {
+public:
+  Http::TestRequestHeaderMapImpl request_headers_{
+      {":method", "GET"},
+      {":path", "/request/path?secret=parameter"},
+      {"x-envoy-original-path", "/original/path?secret=parameter"}};
+  Http::TestResponseHeaderMapImpl response_headers_;
+  Http::TestResponseTrailerMapImpl response_trailers_;
+  StreamInfo::MockStreamInfo stream_info_;
+  std::string body_;
+
+  envoy::config::core::v3::SubstitutionFormatString config_;
+  NiceMock<Server::Configuration::MockFactoryContext> context_;
+};
+
+TEST_F(ReqWithoutQueryTest, TestStripQueryString) {
+  const std::string yaml = R"EOF(
+  text_format_source:
+    inline_string: "%REQ_WITHOUT_QUERY(:PATH)%"
+  formatters:
+    - name: envoy.formatter.req_without_query
+      typed_config:
+        "@type": type.googleapis.com/envoy.extensions.formatter.req_without_query.v3.ReqWithoutQuery
+)EOF";
+  TestUtility::loadFromYaml(yaml, config_);
+
+  auto formatter =
+      ::Envoy::Formatter::SubstitutionFormatStringUtils::fromProtoConfig(config_, context_.api());
```

Comment:
Calling `SubstitutionFormatStringUtils::fromProtoConfig` directly doesn't work?

Reply:
It did not work, this code is in `Envoy::Extensions::Formatter` namespace.  I can add `using` if we want to avoid having long names?

Reply:
I need fully qualified names since I have two namespaces called `Formatter`.  There is `Envoy::Formatter` [here](https://github.com/envoyproxy/envoy/blob/58a13570f3f4dea9bad8b8fa5e1221d7ed5056de/include/envoy/formatter/substitution_formatter.h#L12) and I myself added `Envoy::Extensions::Formatter` causing the overlap.

The reason why I used `Envoy::Extensions::Formatter` was because the category for access log formatter extensions was already set `envoy.formatter` [here](https://github.com/envoyproxy/envoy/blob/58a13570f3f4dea9bad8b8fa5e1221d7ed5056de/include/envoy/formatter/substitution_formatter.h#L130).  I could have used e.g. `Envoy::Extensions::AccessLogFromatters` and `source/extensions/access_log_formatters/` as directory but I thought these should match with the chosen extension category name.

Reply:
I think the namespaces are fine. Does just `Envoy::Formatter::SubstitutionFormatStringUtils::fromProtoConfig` work? (e.g. no leading `::`). It's also fine to add type aliases via`using` judiciously to make these more readable. That's especially true in the tests and the implementation files. For the public headers, consider the [advice in the style guilde](https://google.github.io/styleguide/cppguide.html#Aliases).